lib/provisioning/account.js

```diff
@@ -0,0 +1,308 @@
+}
+
+
+/**
+ * @desc Creates a new sub-account. Any users that have access to all sub-accounts will also automatically have access
+ *       to the new sub-account.
+ * @param name {string} The display name as shown in the management console.
+ * @param cloud_name {string} A case-insensitive cloud name comprised of alphanumeric and underscore characters.
+ *                            Generates an error if the specified cloud name is not unique across all Cloudinary
+ *                            accounts. Note: Once created, the name can only be changed for accounts with fewer than
+ *                            1000 assets.
+ * @param custom_attributes {object} Any custom attributes you want to associate with the sub-account, as a map/hash of
+ *                                   key/value pairs.
+ * @param enabled {boolean} Whether the sub-account is enabled. Default: true
+ * @param base_account {string} The ID of another sub-account, from which to copy all of the following settings:
+ *                              Size limits, Timed limits, and Flags.
+ * @param [options] {object} - See {@link https://cloudinary.com/documentation/cloudinary_sdks#configuration_parameters|Configuration parameters} in the SDK documentation.
+ * @param callback
+ */
+function create_sub_account(name, cloud_name, custom_attributes, enabled, base_account, options = {}, callback) {
+  let params = {
+    cloud_name: cloud_name,
+    name,
+    custom_attributes: custom_attributes,
+    enabled,
+    base_sub_account_id: base_account,
+  };
+
+  options.content_type = "json";
+  let uri = ['sub_accounts'];
+  return call_account_api('POST', uri, params, callback, options);
+}
+
+/**
+ * @desc Deletes the specified sub-account. Supported only for accounts with fewer than 1000 images.
+ * @param sub_account_id {string} - The ID of the sub-account to delete.
+ * @param [options] {object} - See {@link https://cloudinary.com/documentation/cloudinary_sdks#configuration_parameters|Configuration parameters} in the SDK documentation.
+ * @param [callback] {function}
+ */
+function delete_sub_account(sub_account_id, options = {}, callback) {
+  let uri = ['sub_accounts', sub_account_id];
+  return call_account_api('DELETE', uri, {}, callback, options);
+}
+
+/**
+ * @desc Updates the the specified details of the sub-account.
+ * @param sub_account_id {string} - The ID of the sub-account.
+ * @param [name] {string} - The display name as shown in the management console.
+ * @param [cloud_name] {string} - A case-insensitive cloud name comprised of alphanumeric and underscore characters.
+ *                                Use with caution - generates an error if the cloud name is not unique across all
+ *                                Cloudinary accounts.
+ *                                Note: Can only be changed for accounts with fewer than 1000 images.
+ * @param [custom_attributes] {object} - Any custom attributes you want to associate with the sub-account, as a map/hash
```

Comment:
Is "map/hash" correct for node.js?

Reply:
Yea it's understandable enough, all objects in JS are actually a form of hash.